### PR TITLE
[performance doc] Power and Cooling

### PR DESCRIPTION
**Pull request by stas00**
This PR supplies performance doc additions:

- Power and Cooling
- Grad accumulation
- links to various benchmarks I recently created
- a direct link to the scalabilty doc

@LysandreJik, @sgugger

### CODE REVIEW

File: docs/source/performance.mdx [hunk 1]
```diff
@@ -388,10 +430,26 @@ Note: tf32 mode is internal to CUDA and can't be accessed directly via `tensor.t
...
+
+
+### Gradient Accumulation
+
+Besides using gradient accumulation when a larger batch size is needed and can't normally be fit into the gpu's memory, enabling can also provide a 20-30% speed up as it runs the optimizer less often, e.g. see benchmarks for [RTX-3090](https://github.com/huggingface/transformers/issues/14608#issuecomment-1004392537)
+and [A100](https://github.com/huggingface/transformers/issues/15026#issuecomment-1004592231).
+
+To activate this feature in 🤗 Trainer add `--gradient_accumulation_steps 4` to its arguments (experiment with the value to get the best performance.
+
+It's important to remember that using gradient accumulation you may end up with a much larger effective batch size, so you may need to adjust the learning rate, its warm up and for very short datasets it'll impact the loss as the training will end up doing less steps than normal.
```
Comment (sgugger) [inline docs/source/performance.mdx:446]: I don't think this comment is accurate. Using gradient accumulation is the same as using a larger batch size: so you're not getting a speedup because of it (compared to the same effective batch size), it just allows you to run the training at the batch size you want.

File: docs/source/performance.mdx [hunk 2]
```diff
@@ -388,10 +430,26 @@ Note: tf32 mode is internal to CUDA and can't be accessed directly via `tensor.t
...
+
+
 
 ### Gradient Checkpointing
 
-One way to use significantly less GPU memory is to enabled "Gradient Checkpointing" (also known as "activation checkpointing"). When enabled, a lot of memory can be freed at the cost of small decrease in the training speed due to recomputing parts of the graph during back-propagation.
+One way to use significantly less GPU memory is to enabled "Gradient Checkpointing" (also known as "activation checkpointing"). When enabled, a lot of memory can be freed at the cost of small decrease in the training speed due to recomputing parts of the graph during back-propagation. The slow down will depend on the model but quite often it is around 20-30%.
```
Comment (sgugger) [inline docs/source/performance.mdx:452]:
```suggestion
One way to use significantly less GPU memory is to enabled "Gradient Checkpointing" (also known as "activation checkpointing"). When enabled, a lot of memory can be freed at the cost of small decrease in the training speed due to recomputing parts of the graph during back-propagation. The slowdown will depend on the model but quite often it is around 20-30%.
```

File: docs/source/performance.mdx [hunk 3]
```diff
@@ -414,6 +472,10 @@ https://docs.nvidia.com/deeplearning/performance/dl-performance-fully-connected/
 
 For parameters that are small, there is also [Dimension Quantization Effects](https://docs.nvidia.com/deeplearning/performance/dl-performance-matrix-multiplication/index.html#dim-quantization) to consider, this is where tiling happens and the right multiplier can have a significant speedup.
 
+Additionally, as explained in the [Gradient Accumulation](#gradient-accumulation) section, the bigger the batch size the less often the optimizer is run, the faster the training is (considering the same dataset length). See benchmarks
+for [RTX-3090](https://github.com/huggingface/transformers/issues/14608#issuecomment-1004392537)
```
Comment (sgugger) [inline docs/source/performance.mdx:476]: This sentence is correct, so you should use similar prose in the gradient accumulation section.

**Review by sgugger — APPROVED** [review]
Thanks for documenting this!

File: docs/source/performance.mdx (hunk 1 above)
Comment (stas00) [reply docs/source/performance.mdx:446]: could you please be more specific which part(s) you think are inaccurate? The effective batch size?

Comment (sgugger) [reply docs/source/performance.mdx:446]: > enabling can also provide a 20-30% speed up as it runs the optimizer less often

That is not accurate. Using a larger batch size gives the speedup, not gradient accumulation per se.

Comment (stas00) [reply docs/source/performance.mdx:446]: Oh, I see what I mean now. How about changing the first para to:

> Since gradient accumulation essentially is identical to having a larger batch size, just as with the larger batch size here you you are likely to see a 20-30% speed up due to the optimizer running less often.
